web-services/query/src/main/java/datawave/webservice/query/runner/QueryExecutorBean.java

```diff
@@ -488,7 +489,17 @@ private QueryData validateQuery(String queryLogicName, MultivaluedMap<String,Str
             Arrays.sort(dns);
             qd.dnList = Arrays.asList(dns);
             qd.proxyServers = dp.getProxyServers();
+            
+            // Verify that the calling principal has access to the query logic.
+            if (!qd.logic.containsDNWithAccess(qd.dnList)) {
+                UnauthorizedQueryException qe = new UnauthorizedQueryException("User " + qd.userDn + " is not part of the authorized DNs for the query logic "
```

Comment:
This message is a little misleading as we are looking for an intersection with qd.dnList, not just the qd.userDn.  Perhaps something like "None of the DNs used have access to this query logic: " + qd.dnList.

Reply:
Good point. I've modified the exception message as requested.